src/openrct2-ui/audio/AudioChannel.cpp

```diff
@@ -220,14 +181,11 @@ namespace OpenRCT2::Audio
             _deletesourceondone = value;
         }
 
-        bool IsPlaying() const override
-        {
-            return !_done;
-        }
+        [[nodiscard]] bool IsPlaying() const override { return !_done; }
 
         void Play(IAudioSource* source, int32_t loop) override
         {
-            _source = static_cast<ISDLAudioSource*>(source);
+            _source = dynamic_cast<ISDLAudioSource*>(source);
```

Comment:
Whats the reason for dynamic_cast here? Its a much heavier operation than static_cast

Reply:
Theoretically one might pass a non-ISDLAudioSource in Play which is not what we want. 
The `dynamic_cast` sets `_source` to a `nullptr` when `source` is not a `ISDLAudioSource`. 
However, I guess we know that an `ISDLAudioSource` is passed? In that case we could keep the `static_cast` for performance and drop the little extra provided safety.

Reply:
Yeah I don't think its worth it, highly doubt anyone will ever swap it.

Reply:
Let's change this back then.